detox/src/ipc/IPCClient.js

```diff
@@ -59,27 +59,27 @@ class IPCClient {
     const sessionState = await this._emit('reportTestResults', {
       testResults: testResults.map(r => serializeObjectWithError(r, 'testExecError')),
     });
-    this._state.patch(sessionState);
+    this._sessionState.patch(sessionState);
   }
 
   async _connectToServer() {
     const serverId = this.serverId;
 
     this._serverConnection = await new Promise((resolve, reject) => {
-      this._client.connectTo(serverId, (client) => {
+      this._ipc.connectTo(serverId, (client) => {
         client.of[serverId]
           .on('error', reject)
           .on('disconnect', () => reject(new DetoxInternalError('IPC server has unexpectedly disconnected.')))
           .on('connect', () => resolve(client.of[serverId]));
       });
     });
 
-    this._serverConnection.on('sessionStateUpdate', this._onSessionStateUpdate);
+    this._serverConnection.on('', this._onSessionStateUpdate);
```

Comment:
Why did you decide to remove the event name?

Reply:
Sorry, ignore that 😓  that's an error. Already force pushed with the fix